src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,154 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (size == 0) {
+            newNode = new Node<>(null, value, null);
+            head = newNode;
+        } else {
+            newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> nodeToAdd;
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("index: " + index + " Out of bound size : " + size);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> node = findNodeByIndex(index);
+        if (index == 0) {
+            nodeToAdd = new Node<>(null, value, node);
+            head = nodeToAdd;
+        } else {
+            nodeToAdd = new Node<>(node.prev, value, node);
+            node.prev.next = nodeToAdd;
+        }
+        node.prev = nodeToAdd;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T listElement : list) {
+            add(listElement);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> node = findNodeByIndex(index);
+        return node.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = findNodeByIndex(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = findNodeByIndex(index);
+        unlink(node,index);
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        boolean result = false;
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if (node.value == object || node.value != null && node.value.equals(object)) {
+                result = true;
+                remove(i);
+                break;
```

Comment:
```suggestion
                return remove(i);
```

Reply:
but return type in method remove(by index) is <T>
i will get a ClassCastException in method remove(by <T> Object) if that return type is boolean?
how i can fix that?)